assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,328 @@
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				expect( action.next().value.type ).toEqual( 'START_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FETCH_SAVE_SOME_DATA' );
+				expect( action.next( 42 ).value.type ).toEqual( 'RECEIVE_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FINISH_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value ).toEqual( {
+					response: 42,
+					error: undefined,
+				} );
+			} );
+
+			it( 'yields the expected actions for an error request', () => {
+				const fetchStoreDefinition = createFetchStore( {
+					baseName: 'SaveSomeData',
+					controlCallback: async () => true,
+					reducerCallback: ( state ) => state,
+				} );
+
+				const action = fetchStoreDefinition.actions.fetchSaveSomeData();
+
+				const error = {
+					code: 'this-went-wrong',
+					message: 'This went wrong.',
+				};
+
+				expect( action.next().value.type ).toEqual( 'START_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value.type ).toEqual( 'FETCH_SAVE_SOME_DATA' );
+				expect( action.throw( error ).value.type ).toEqual( 'CATCH_FETCH_SAVE_SOME_DATA' );
+				expect( action.next().value ).toEqual( {
+					response: undefined,
+					error,
+				} );
+			} );
+
+			it( 'makes a network request based on controlCallback', async () => {
+				const expectedResponse = 'response-value';
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( expectedResponse ),
+						{ status: 200 }
+					);
+
+				const { response, error } = await dispatch.fetchGetSomeData( {}, 'value-to-key-response-by' );
+
+				expect( error ).toEqual( undefined );
+				expect( response ).toEqual( expectedResponse );
+				expect( store.getState().data ).toEqual( {
+					'value-to-key-response-by': expectedResponse,
+				} );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const errorResponse = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( errorResponse ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				const { response, error } = await dispatch.fetchGetSomeData( {}, 'value-to-key-response-by' );
+
+				expect( error ).toEqual( errorResponse );
+				expect( response ).toEqual( undefined );
+				expect( store.getState().data ).toEqual( undefined );
+			} );
+
+			it( 'sets flag for request being in progress', async () => {
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/test\/data\/some-data/
+					)
+					.mockResponseOnce(
+						JSON.stringify( { someValue: 42 } ),
```

Comment:
If the response is arbitrary, let's annotate that with a comment so we're not looking for `someValue` or 42 in the test 😄